cmd/integration_test.go

```diff
@@ -306,85 +328,109 @@ func TestSmokescreenIntegration(t *testing.T) {
+						TargetHost:   host,
+						TargetScheme: targetScheme,
+						RoleName:     generateRoleForAction(action),
 					}
 					testCases = append(testCases, testCase)
 				}
 			}
 		}
 
 		baseCase := TestCase{
-			OverConnect: overConnect,
-			ProxyURL:    servers[false].URL,
-			TargetPort:  outsideListenerPort,
+			OverConnect:  overConnect,
+			ProxyURL:     servers[false].URL,
+			TargetPort:   outsideListenerPort,
+			TargetScheme: "http",
 		}
 
+		// Empty roles should default deny per the test config
 		noRoleDenyCase := baseCase
-		noRoleDenyCase.Host = "127.0.0.1"
+		noRoleDenyCase.TargetHost = "127.0.0.1"
 		noRoleDenyCase.ExpectAllow = false
 
-		noRoleAllowCase := baseCase
-		noRoleAllowCase.Host = "localhost"
-		noRoleAllowCase.ExpectAllow = true
-
-		unknownRoleDenyCase := noRoleDenyCase
+		// Unknown roles should default deny per the test config
+		unknownRoleDenyCase := baseCase
+		unknownRoleDenyCase.TargetHost = "127.0.0.1"
 		unknownRoleDenyCase.RoleName = "unknown"
+		unknownRoleDenyCase.ExpectAllow = false
 
-		unknownRoleAllowCase := noRoleAllowCase
-		unknownRoleAllowCase.RoleName = "unknown"
-
-		badIPRangeCase := baseCase
 		// This must be a global unicast, non-loopback address or other IP rules will
 		// block it regardless of the specific configuration we're trying to test.
-		badIPRangeCase.Host = "1.1.1.1"
+		badIPRangeCase := baseCase
+		badIPRangeCase.TargetHost = "1.1.1.1"
 		badIPRangeCase.ExpectAllow = false
 		badIPRangeCase.RoleName = generateRoleForAction(acl.Open)
 
-		badIPAddressCase := baseCase
 		// This must be a global unicast, non-loopback address or other IP rules will
 		// block it regardless of the specific configuration we're trying to test.
-		badIPAddressCase.Host = "1.0.0.1"
+		badIPAddressCase := baseCase
+		badIPAddressCase.TargetHost = "1.0.0.1"
 		badIPAddressCase.TargetPort = 123
 		badIPAddressCase.ExpectAllow = false
 		badIPAddressCase.RoleName = generateRoleForAction(acl.Open)
 
-		proxyCase := baseCase
 		// We expect this URL to always return a non-200 status code so that
 		// this test will fail if we're not respecting the UpstreamProxy setting
 		// and instead going straight to this host.
-		proxyCase.Host = "aws.s3.amazonaws.com"
-		proxyCase.UpstreamProxy = outsideListenerUrl.String()
-		proxyCase.ExpectAllow = true
-		proxyCase.RoleName = generateRoleForAction(acl.Open)
+		if overConnect {
+			proxyCase := baseCase
+			proxyCase.TargetScheme = "https"
+			proxyCase.TargetHost = "aws.s3.amazonaws.com"
+			proxyCase.TargetPort = 443
+			proxyCase.UpstreamProxy = outsideListenerUrl.String()
+			proxyCase.ExpectAllow = true
+			proxyCase.RoleName = generateRoleForAction(acl.Open)
+			testCases = append(testCases, &proxyCase)
+		}
 
 		testCases = append(testCases,
-			&unknownRoleAllowCase, &unknownRoleDenyCase,
-			&noRoleAllowCase, &noRoleDenyCase,
+			&unknownRoleDenyCase, &noRoleDenyCase,
 			&badIPRangeCase, &badIPAddressCase,
-			&proxyCase,
 		)
 	}
 
```

Comment:
For the test cases below here, it would be good to use a meaningful name instead of `""` as the first argument to `t.Run()`. How about adding a `TestCase.String()` method that encodes test case parameters in a readable way?

Reply:
Prior to every test running we print out the test case parameters. Because we can't generate meaningful `TestCase.String()` output for the autogenerated tests I'm inclined to leave this as is for now.